common/src/state/pending_message.rs

```diff
@@ -1,17 +1,44 @@
-use std::collections::HashMap;
+use std::{collections::HashMap, path::PathBuf};
 
 use uuid::Uuid;
-use warp::{constellation::Progression, crypto::DID};
+use warp::{constellation::Progression, crypto::DID, raygun::Location};
 
 use crate::warp_runner::ui_adapter::Message;
 // We can improve message equality detection if warp e.g. can send us their assigned uuid.
 // Else it is just a guesswork
 #[derive(Clone, Debug)]
 pub struct PendingMessage {
-    pub attachments_progress: HashMap<String, FileProgression>,
+    pub attachments_progress: HashMap<FileLocation, FileProgression>,
     pub message: Message,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum FileLocation {
```

Comment:
Assuming this is done because `Hash` is not implemented in `Location`, a PR on warp side does add this so this might be redundant in the neat future

Reply:
ye that was the reason for it